kusto/ingest/internal/conn/conn_test.go

```diff
@@ -118,58 +124,61 @@ func TestStream(t *testing.T) {
 	conn.inTest = true
 
 	for _, test := range tests {
-		ctx, _ := context.WithTimeout(context.Background(), 3*time.Second)
-
-		b, err := json.Marshal(test.payload)
-		if err != nil {
-			panic(err)
-		}
-
-		var payload bytes.Buffer
-		zw := gzip.NewWriter(&payload)
-		_, err = zw.Write(b)
-		if err != nil {
-			panic(err)
-		}
-
-		if err := zw.Close(); err != nil {
-			panic(err)
-		}
-
-		err = conn.Write(ctx, "database", "table", &payload, properties.JSON, test.mappingName)
-
-		switch {
-		case err == nil && test.err:
-			t.Errorf("TestStream(%s): got err == nil, want err != nil", test.desc)
-			continue
-		case err != nil && !test.err:
-			t.Errorf("TestStream(%s): got err == %s, want err == nil", test.desc, err)
-			continue
-		case err != nil:
-			continue
-		}
-
-		switch {
-		case server.req.Header.Get("Content-Type") != "application/json; charset=utf-8":
-			t.Fatalf("TestStream(%s): Content-Type: got %s, want %s", test.desc, server.req.Header.Get("Content-Type"), "application/json; charset=utf-8")
-		case server.req.URL.Query().Get("streamFormat") != "Json":
-			t.Fatalf("TestStream(%s): Query Variable(streamFormat): got %s, want json", test.desc, server.req.URL.Query().Get("streamFormat"))
-		case server.req.URL.Query().Get("mappingName") != test.mappingName:
-			t.Fatalf("TestStream(%s): Query Variable(mappingName): got %s, want %s", test.desc, server.req.URL.Query().Get("mappingName"), test.mappingName)
-		}
-
-		if !strings.HasPrefix(server.req.Header.Get("x-ms-client-request-id"), "KGC.execute;") {
-			t.Fatalf("TestStream(%s): x-ms-client-request-id(%s): was not expected format", test.desc, server.req.Header.Get("x-ms-client-request-id"))
-		}
-		uuid.MustParse(strings.TrimPrefix(server.req.Header.Get("x-ms-client-request-id"), "KGC.execute;"))
-
-		got := fakeContent{}
-		if err := json.Unmarshal(server.out, &got); err != nil {
-			t.Fatalf("TestStream(%s): could not unmarshal data sent to server: %s", test.desc, err)
-		}
-
-		if diff := pretty.Compare(test.payload, got); diff != "" {
-			t.Fatalf("TestStream(%s) -want/+got:\n%s", test.desc, diff)
-		}
+		func() {
+			ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+			defer cancel()
+
+			b, err := json.Marshal(test.payload)
+			if err != nil {
+				panic(err)
+			}
+
+			var payload bytes.Buffer
+			zw := gzip.NewWriter(&payload)
+			_, err = zw.Write(b)
+			if err != nil {
+				panic(err)
+			}
+
+			if err := zw.Close(); err != nil {
+				panic(err)
+			}
+
+			err = conn.Write(ctx, "database", "table", &payload, properties.JSON, test.mappingName)
+
+			switch {
+			case err == nil && test.err:
+				t.Errorf("TestStream(%s): got err == nil, want err != nil", test.desc)
+				return
```

Comment:
You sure Go supports multiple errors in the same test method?

Reply:
I don't understand.
It does display multiple errors if there are